modules/data.land/R/extract_soil_nc.R

```diff
@@ -4,91 +4,191 @@
 #' @param lat Latitude 
 #' @param lon Longitude
 #' @param size Ensemble size
-#' @param radius radius in meters is used to take soil type samples around the site
+#' @param grid_size Size of the spatial sampling grid (default: 3)
+#' @param grid_spacing Spacing between grid cells in meters (default: 10)
 #' @param depths  Standard set of soil depths in m to create the ensemble of soil profiles with.
 #'
 #' @return It returns the address for the generated soil netcdf file
 #'
 #' @importFrom rlang .data
-#'
+#' @importFrom sf st_as_sf st_transform
+#' @importFrom terra rast crds
 #' @examples
 #' \dontrun{
 #'    outdir  <- "~/paleon/envTest"
 #'    lat     <- 40
 #'    lon     <- -80
 #'    PEcAn.data.land::extract_soil_gssurgo(outdir, lat, lon)
 #' }
-#' @author Hamze Dokoohaki
+#' @author Hamze Dokoohaki and Akash
 #' @export
 #' 
-extract_soil_gssurgo<-function(outdir, lat, lon, size=1, radius=500, depths=c(0.15,0.30,0.60)){
+extract_soil_gssurgo<-function(outdir, lat, lon, size=1, grid_size=3, grid_spacing=10, depths=c(0.15,0.30,0.60)){
   # I keep all the ensembles here 
   all.soil.ens <-list()
   
-  # I ask the gSSURGO to find all the mukeys (loosely can be thought of soil type) within 500m of my site location. 
-  # Basically I think of this as me going around and taking soil samples within 500m of my site.
-  #https://sdmdataaccess.nrcs.usda.gov/SpatialFilterHelp.htm
-  mu.Path <- paste0(
-    "https://sdmdataaccess.nrcs.usda.gov/Spatial/SDMWGS84Geographic.wfs?",
-    "SERVICE=WFS",
-    "&VERSION=1.1.0",
-    "&REQUEST=GetFeature&TYPENAME=MapunitPoly",
-    "&FILTER=",
-      "<Filter>",
-        "<DWithin>",
-          "<PropertyName>Geometry</PropertyName>",
-          "<gml:Point>",
-            "<gml:coordinates>", lon, ",", lat, "</gml:coordinates>",
-          "</gml:Point>",
-          "<Distance%20units=%27m%27>", radius, "</Distance>",
-        "</DWithin>",
-      "</Filter>",
-    "&OUTPUTFORMAT=XMLMukeyList"
+  # Grid-based spatial sampling (via WFS queries)
+  proj_crs <- sf::st_crs("+proj=aea +lat_1=29.5 +lat_2=45.5 +lat_0=23 +lon_0=-96 +x_0=0 +y_0=0 +ellps=GRS80 +datum=NAD83 +units=m +no_defs")
+  wgs84_crs <- sf::st_crs(4326)
+  
+  # Convert center lat/lon to projected coordinates
+  point_sf <- sf::st_sfc(sf::st_point(c(lon, lat)), crs = wgs84_crs)
+  point_proj <- sf::st_transform(point_sf, proj_crs)
+  coords_proj <- sf::st_coordinates(point_proj)
+  
+  # Define grid extent 
+  half_extent <- (grid_size - 1) / 2 * grid_spacing
+  xmin <- coords_proj[1] - half_extent
+  xmax <- coords_proj[1] + half_extent
+  ymin <- coords_proj[2] - half_extent
+  ymax <- coords_proj[2] + half_extent
+  
+  # Create raster template
+  raster_template <- terra::rast(
+    xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax,
+    resolution = grid_spacing, crs = proj_crs
```

Comment:
This all seems like a really long way around to me - why this rather than define a buffer around the points and then extract cells that fall inside it?

Reply:
Perhaps something like:

```suggestion
point <- terra::vect(cbind(lon, lat), crs = wgs84_crs) |>
  terra::project(proj_crs)

half_extent <- (grid_size - 1) / 2 * grid_spacing
ext <- terra::ext(point)
ext <- ext + c(-half_extent, half_extent, -half_extent, half_extent)

raster_template <- terra::rast(ext, resolution = grid_spacing, crs = proj_crs)
```

?

Reply:
@infotroph If I'm understanding correctly you are requesting replacing lines 41-50 with something along the lines of: 

```suggestion
  bb <- point_proj |>
    sf::st_buffer(dist = half_extent) |>
    sf::st_bbox()

  raster_template <- terra::rast(
    xmin = bb["xmin"], xmax = bb["xmax"],
    ymin = bb["ymin"], ymax = bb["ymax"],
    resolution = grid_spacing, crs = proj_crs
  )
```

But this seems like more of a stylistic than a functional change: it doesn't reduce lines of code or noticeably clarify intent. Am I am missing something?

Reply:
After a digging into this grid vs buffer, I am convinced to proceed with current grid based approach due few reasons:
The main trade-off is precision vs simplicity - the current method guarantees exact 10m spacing in projected coordinates, while a buffer approach would be less code but with approximate spacing.  However, the grid approach uses multiple simple WFS point queries (`DWithin` filters) that are server-optimized and fault-tolerant, while buffer approach requires a single complex polygon query that's computationally expensive and fails catastrophically.

Reply:
Closing this thread because what's here works and I think the next refactor should completely drop the grid sampling approach, but for the record I was thinking all of lines 43 - 59 could be reduced to basically `data.frame(lon=lon, lat=lat) |> terra::vect(crs = "EPSG:4326") |> terra::buffer(radius) |> terra::rast(nrow = grid_size, ncol = grid_size) |> terra::crds()`